src/utils/context.ts

```diff
@@ -0,0 +1,247 @@
+* Eclipse Public License v2.0 which accompanies this distribution, and is available at *
+* https://www.eclipse.org/legal/epl-v20.html                                      *
+*                                                                                 *
+* SPDX-License-Identifier: EPL-2.0                                                *
+*                                                                                 *
+* Copyright Contributors to the Zowe Project.                                     *
+*                                                                                 *
+*/
+
+import * as extension from "../extension";
+import { TreeItem } from "vscode";
+
+/**
+ *
+ * The contextValue is made up with the name of the node type as the first value in the sequence.
+ * All subsequent attributes of the contextValue are preceded by an underscore character.
+ * example:
+ * pds_fav represents a pds file indicated as a favorite.
+ *
+ * Additional fields can be added as underscore values
+ * example:
+ * job_fav_rc=CC 0000
+ *
+ */
+
+/**
+ * Helper function which identifies if the node is a job
+ * @param node
+ * @return true if a job, false otherwise
+ */
+export function isJob(node: TreeItem): boolean {
+    return new RegExp("^" + extension.JOBS_JOB_CONTEXT ).test(node.contextValue);
+}
+
+/**
+ * Helper function which identifies if the node is a pds or ds and a favorite
+ * @param node
+ * @return true if a favorite pds, ds, false otherwise
+ */
+export function isFavoritePsDs(node: TreeItem): boolean {
+    return new RegExp("^(" + extension.DS_PDS_CONTEXT + "|" +
+                    extension.DS_DS_CONTEXT + ")(.*" + extension.FAV_SUFFIX + ")").test(node.contextValue);
+}
+
+/**
+ * Helper function which identifies if the node is a ds and a favorite
+ * @param node
+ * @return true if a favorite ds, false otherwise
+ */
+export function isFavoriteDs(node: TreeItem): boolean {
+    return new RegExp("^(" + extension.DS_DS_CONTEXT + ")(.*" + extension.FAV_SUFFIX + ")").test(node.contextValue);
+}
+
+/**
+ * Helper function which identifies if the node is a job and it's a favorite
+ * @param node
+ * @return true if a favorite job, false otherwise
+ */
+export function isFavoriteJob(node: TreeItem): boolean {
+    return new RegExp("^(" + extension.JOBS_JOB_CONTEXT + ")(.*" + extension.FAV_SUFFIX + ")").test(node.contextValue);
+}
+
+/**
+ * Helper function which identifies if the node is a pds and a favorite
+ * @param node
+ * @return true if a favorite pds, false otherwise
+ */
+export function isFavoritePds(node: TreeItem): boolean {
+    return new RegExp("^(" + extension.DS_PDS_CONTEXT + ")(.*" + extension.FAV_SUFFIX + ")").test(node.contextValue);
+}
+
+/**
+ * Helper function which identifies if the node is a Favorite binary or text file
+ * @param node
+ * @return true if a Favorite binary or text file, false otherwise
+ */
+export function isFavoriteTextorBinary(node: TreeItem): boolean {
```

Comment:
Should this have a capital `O` for `Or`

Reply:
I'll fix that. Helps me remember to do it next time.